### PR TITLE
[ResourceBundle] Abstract Resource Configuration

### PR DESCRIPTION
**Pull request by Strontium-90**
| Q             | A
| ------------- | ---
| Bug fix?      | [no]
| New feature?  | [yes]
| BC breaks?    | [no]
| Deprecations? | [no]
| Tests pass?   | [yes]
| Fixed tickets | [ ]
| License       | MIT
| Doc PR        | []

Helper for configuring resource bundles.
Example of usage included.

### CODE REVIEW

File: src/Sylius/Bundle/ResourceBundle/DependencyInjection/Driver/DoctrineODMDriver.php [hunk 1]
```diff
@@ -34,9 +35,9 @@ public function getSupportedDriver()
      */
     protected function getRepositoryDefinition(array $classes)
     {
-        $repositoryClass = 'Sylius\Bundle\ResourceBundle\Doctrine\ODM\MongoDB\DocumentRepository';
+        $repositoryClass = new Parameter('sylius.mongodb-odm.repository.class');
```
Comment (stloyd) [inline src/Sylius/Bundle/ResourceBundle/DependencyInjection/Driver/DoctrineODMDriver.php]: I'm not sure about mixing `-` & `.`, maybe better would be to use `_` instead?

File: src/Sylius/Bundle/ResourceBundle/DependencyInjection/Driver/DoctrineODMDriver.php [hunk 2]
```diff
@@ -34,9 +35,9 @@ public function getSupportedDriver()
...
 
-        if (isset($classes['repository'])) {
+        if (!empty($classes['repository'])) {
```
Comment (stloyd) [inline src/Sylius/Bundle/ResourceBundle/DependencyInjection/Driver/DoctrineODMDriver.php]: Why? `isset()` is enough for `null` values, and this one can't be an empty string anyway...

File: src/Sylius/Bundle/ResourceBundle/Tests/DependencyInjection/AbstractResourceConfigurationTest.php [hunk 3]
```diff
@@ -0,0 +1,316 @@
...
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Sylius\Bundle\ResourceBundle\Tests\DependencyInjection;
+
+use ReflectionMethod;
```
Comment (stloyd) [inline src/Sylius/Bundle/ResourceBundle/Tests/DependencyInjection/AbstractResourceConfigurationTest.php]: Instead just call `\ReflectionMethod` where it's needed.

File: src/Sylius/Bundle/ResourceBundle/Tests/DependencyInjection/AbstractResourceConfigurationTest.php [hunk 4]
```diff
@@ -0,0 +1,316 @@
...
+                    'driver' => ''
+                )
+            )
+        );
+    }
+
+    /**
+     * @param NodeDefinition $definition
+     *
+     * @return \Symfony\Component\Config\Definition\Builder\NodeInterface|\Symfony\Component\Config\Definition\NodeInterface
```
Comment (stloyd) [inline src/Sylius/Bundle/ResourceBundle/Tests/DependencyInjection/AbstractResourceConfigurationTest.php]: Add `use` instead of FQCN.

**Comment by aRn0D** [comment]
@pjedrzejewski what is your opinion about my last comment (21 Sep 2014) ?

**Comment by pjedrzejewski** [comment]
IMHO, this is an overkill. I like simplifying things, but here I do not see much gain in code/duplication reduction. We lose the flexibility as every configuration can be different.

**Pull request closed without merging**